python/cudf/cudf/core/column/string.py

```diff
@@ -1786,6 +1791,66 @@ def translate(self, table, **kwargs):
             cpp_translate(self._column, table), **kwargs
         )
 
+    def normalize_spaces(self):
+        return cpp_normalize_spaces(self._column)
+
+    def tokenize(self, delimiter=" "):
+        delimiter = _massage_string_arg(delimiter, "delimiter", allow_col=True)
+        return cpp_tokenize(self._column, delimiter)
+
+    def token_count(self, delimiter=" "):
+        delimiter = _massage_string_arg(delimiter, "delimiter", allow_col=True)
+        return cpp_count_tokens(self._column, delimiter)
+
+    def ngrams(self, n=2, separator="_"):
+        separator = _massage_string_arg(separator, "separator")
+        return cpp_generate_ngrams(self._column, n, separator)
+
+    def ngrams_tokenize(self, n=2, delimiter=" ", separator="_"):
+        delimiter = _massage_string_arg(delimiter, "delimiter")
+        separator = _massage_string_arg(separator, "separator")
+
+        return cpp_ngrams_tokenize(self._column, n, delimiter, separator)
+
+
+def _massage_string_arg(value, name, allow_col=False):
+    from cudf._libxx.scalar import Scalar
+    from cudf._libxx.column import Column
+    from cudf.utils.dtypes import is_string_dtype
+
+    allowed_types = ["Scalar"]
+
+    if isinstance(value, str):
+        return Scalar(value, dtype="str")
+
+    if isinstance(value, Scalar) and is_string_dtype(value.dtype):
+        return value
+
+    if allow_col:
+        allowed_types += ["Column"]
+
+        if isinstance(value, list):
+            return column.as_column(value, dtype="str")
+
+        if isinstance(value, Column) and is_string_dtype(value.dtype):
+            return value
+
+    raise ValueError(
+        "Expected {} for {} but got {}".format(
+            _expected_types_format(allowed_types), name, type(value)
+        )
+    )
+
+
+def _expected_types_format(types):
+    if len(types) == 0:
+        raise ValueError
```

Comment:
Can we add a more descriptive exception message here or just get rip of this check since it's deep within internal code anyway?